nx-X11/programs/Xserver/dix/window.c

```diff
@@ -946,10 +945,19 @@ DeleteWindow(void * value, XID wid)
 	if (pWin->prevSib)
 	    pWin->prevSib->nextSib = pWin->nextSib;
     }
+
+#ifdef NXAGENT_SERVER
+    if (pWin -> optional &&
+            pWin -> optional -> colormap &&
+                pWin -> parent)
+    {
+      nxagentSetInstalledColormapWindows(pWin -> drawable.pScreen);
+    }
+#endif
+
```

Comment:
This code portion is GPL-2 while the rest of dix/window.c is MIT (or some such). Move that fragment into a function in the DDX and call that function from here.